specification/communication/data-plane/Microsoft.CommunicationServicesChat/preview/2020-11-01-preview3/communicationserviceschat.json

```diff
@@ -0,0 +1,1589 @@
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "429": {
+            "description": "Too many requests",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Add participants": {
+            "$ref": "./examples/Participants_AddChatParticipants.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/participants/{chatParticipantId}": {
+      "delete": {
+        "tags": [
+          "Participants"
+        ],
+        "summary": "Remove a participant from a thread.",
+        "operationId": "RemoveChatParticipant",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "chatThreadId",
+            "description": "Thread id to remove the participant from.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "chatParticipantId",
+            "description": "Id of the thread participant to remove from the thread.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Request successful."
+          },
+          "401": {
+            "description": "Unauthorized",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "403": {
+            "description": "Forbidden",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "429": {
+            "description": "Too many requests",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Remove participant": {
+            "$ref": "./examples/Participants_RemoveChatParticipant.json"
+          }
+        }
+      }
+    },
+    "/chat/threads": {
+      "post": {
+        "tags": [
+          "Threads"
+        ],
+        "summary": "Creates a chat thread.",
+        "operationId": "CreateChatThread",
```

Comment:
It is good that we have added "Chat" to the types (eg. `ChatMessage`, `ChatThread`, etc.), but for the method names, it becomes redundant: `ChatClient.CreateChatThread`. I think we should remove the duplicated "Chat" from the operation ids so we would have `ChatClient.CreateThread`

Reply:
This one was discussed during private preview (for the method names included), and 3 of the architects were strong about adding Chat, so the decision was to with the prefix across the board. Their argument was that CreateThread or GetThread was too generic and Thread had a strong connotation in software.
Also just to be clear, the 'public' methods on the Clients are not driven by the swagger. They are all hand crafted.